src/domain/model.ts

```diff
@@ -20,10 +20,6 @@ export type AdoptablePets<Location, Pets> = {
   readonly pets: Pets
 }
 
-export type GetPets<Location, Pets> = {
-  getPets(l: Location): Promise<Pets>
-}
+export type GetPets<Env, Location, Pets> = (e: Env, l: Location) => Promise<Pets>
```

Comment:
The generics here are a little distracting, and I'm not sure what problem they are trying to solve.  Any context?  It seems like `GetPets` is a very specific implementation

Reply:
Sorry for the distractions.  I've been exploring some other ideas, one of which is parametric interfaces, similar to [ML signatures and modules](https://en.wikipedia.org/wiki/Standard_ML#Module_system).  They enable writing generic algorithms that work without knowledge of the structure of the types.  In order to stay focused on the ddd+onion stuff here, it's probably best for me just to go with concrete interfaces ... at least until I figure out whether/how parametric interfaces fit in 😄

Reply:
There's an interesting case in the GetLocation interface: UserInfo.  Parameterizing the UserInfo type allows the _application_ layer to decide the type for which it will support getting location information, without the domain layer having to know about it.